datastore/mongodb/filters.go

```diff
@@ -40,3 +40,18 @@ func (c *Client) CreateFilter(ctx context.Context, f *model.Filter) error {
 
 	return nil
 }
+
+// GetFilterDimensions gets the dimensions for a Filter in the CantabularFilters collection
+func (c *Client) GetFilterDimensions(ctx context.Context, fID string) ([]model.Dimension, error) {
+	var err error
+
+	col := c.collections.filters
+
+	var f model.Filter
+
+	if err = c.conn.Collection(col.name).FindOne(ctx, bson.M{"id": fID}, f); err != nil {
+		return []model.Dimension{}, errors.Wrap(err, "failed to get filter")
```

Comment:
You can return `nil` here rather than `[]model.Dimension{}`.

Reply:
I think our dp-mongodb has an error type we can check against to determine if it's NotFound error `https://github.com/ONSdigital/dp-mongodb/blob/main/mongodb/error.go`. Similar to what's done in collections.go lock() function you can add some extra detail to the error here by using the package's error tpye and doing something like:
```
	if err != nil {
		err := &er{
			err: errors.Wrap(err, "failed to get filter"),
		}
		if errors.Is(err, dpmongo.ErrNoDocumentFound) {
			err.notFound = true
		}
        }
```
This will make the statusCode assert to 404 in the handler